### ISSUE TITLE
Errors are swallowed when using fromPromise(A+ promise)

### ISSUE THREAD

**Issue by AlexGalays** [baconjs/bacon.js#511]
When in dev mode, it would be very useful to have a global, catch-all error handler since bacon swallow all errors. A tiny error in a map() somewhere downstream, and nothing happens, no stream is consumed, no error is thrown.

It could be used as follow:

```
if (process.env.dev)
  Bacon.onError = e -> console.error(e);
```

**Comment by raimohanska** [comment]
No it doesnt swallow all errors does it!? There was a bug but with latest version there should not be such behavior

**Comment by raimohanska** [comment]
true! needs investigation.

**Comment by raimohanska** [comment]
oh, it's Q that's swallowing the exception, not Bacon :)

**Comment by AlexGalays** [comment]
Yes and no :p

Q or any A+ promise implementation will by design swallow an exception in a `then` success handler if no other action is taken (in the case of Q, you would have to use `done()` to tell it the promise chain is over and all errors have been handled).

Since bacon only relies on the presence of a 'then' function, I think the only action possible is for it to add a try-catch that call the same thing as the error handler.

**Comment by raimohanska** [comment]
This is duplicate to #501

**Comment by raimohanska** [comment]
Fixed in 0.7.50
